x-pack/plugins/lens/public/indexpattern_plugin/loader.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Chrome } from 'ui/chrome';
+import { ToastNotifications } from 'ui/notify/toasts/toast_notifications';
+import { IndexPatternField } from './indexpattern';
+
+interface IndexPatternSavedObject {
+  id: string;
+  type: string;
+  attributes: {
+    title: string;
+    timeFieldName?: string;
+    fields: string;
+    fieldFormatMap: string;
+  };
+}
+
+export const getIndexPatterns = (chrome: Chrome, toastNotifications: ToastNotifications) => {
+  const savedObjectsClient = chrome.getSavedObjectsClient();
+  return savedObjectsClient
+    .find({
+      type: 'index-pattern',
+      perPage: 1000, // TODO: Paginate index patterns
+    })
+    .then(resp => {
+      return resp.savedObjects.map(savedObject => {
+        const { id, attributes } = (savedObject as unknown) as IndexPatternSavedObject;
```

Comment:
Actually the saved object client is strictly typed. This can be used by a type attribute on the find method:
```
interface IndexPatternAttributes extends SavedObjectAttributes {
  title: string;
  timeFieldName: string | null;
  fields: string;
  fieldFormatMap: string;
}

export const getIndexPatterns = (chrome: Chrome, toastNotifications: ToastNotifications) => {
  const savedObjectsClient = chrome.getSavedObjectsClient();
  return savedObjectsClient
    .find<IndexPatternAttributes>({
      type: 'index-pattern',
      perPage: 1000, // TODO: Paginate index patterns
    })
    .then(resp => {
      return resp.savedObjects.map(savedObject => {
        const { id, attributes } = savedObject;
```

Then we don't need to do the unsafe `as unknown as something` and the interface gets shorter.

Reply:
I tried to write the code you suggested and had issues. I didn't know about the `.find<IndexPatternAttributes>(` syntax